extensions/blocks/social-previews/utils.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { has } from 'lodash';
+
+/**
+ * Gets the URL of the media. Tries loading a smaller size available to be used as a thumbnail and falls back to the full size.
+ *
+ * Source: https://github.com/WordPress/gutenberg/blob/HEAD/packages/editor/src/components/post-featured-image/index.js
+ *
+ * @param {object} media
+ * @param {number} currentPostId
+ * @returns {?string} URL address
+ */
+export function getMediaSourceUrl( media, currentPostId ) {
+	if ( ! media ) {
+		return null;
+	}
+
+	const mediaSize = applyFilters(
+		'editor.PostFeaturedImage.imageSize',
+		'post-thumbnail',
+		media.id,
+		currentPostId
+	);
+	if ( has( media, [ 'media_details', 'sizes', mediaSize ] ) ) {
+		// use mediaSize when available
+		return media.media_details.sizes[ mediaSize ].source_url;
+	}
+
+	// get fallbackMediaSize if mediaSize is not available
+	const fallbackMediaSize = applyFilters(
+		'editor.PostFeaturedImage.imageSize',
+		'thumbnail',
+		media.id,
+		currentPostId
+	);
+	if ( has( media, [ 'media_details', 'sizes', fallbackMediaSize ] ) ) {
```

Comment:
Any benefit to `has` here as well?

Reply:
No idea, this was taken verbatim from Gutenberg. I can see if I can update it but I'd rather have as little changes as possible

Reply:
Updated